### PR TITLE
CB-70: Add User model

### PR DESCRIPTION
**Pull request by grabomarcin**
https://team-1626248039387.atlassian.net/browse/CB-70

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# Model description or :nodoc:
+
+class User < ApplicationRecord
+  # == Constants ============================================================
+
+  # == Attributes ===========================================================
+
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
Comment (PatrykTyszka) [inline app/models/user.rb:14]: confirmable is a nice feature so we can add this as well ;) or you can add this later.

File: db/migrate/20210806120522_add_nickname_to_user.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddNicknameToUser < ActiveRecord::Migration[6.1]
+  def change
+    add_column :users, :nickname, :string, null: false
```
Comment (PatrykTyszka) [inline db/migrate/20210806120522_add_nickname_to_user.rb:3]: why did you add this as a separate migration?

**Review by PatrykTyszka — APPROVED** [review]

File: Gemfile [hunk 3]
```diff
@@ -52,3 +52,5 @@ end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
 gem 'tzinfo-data', platforms: %i[mingw mswin x64_mingw jruby]
+
+gem 'devise'
```
Comment (mirecki) [inline Gemfile:56]: dodaj minimalna wersje w gemfile i dodaj to analogicznie do pozostałych i.e jak ma być na test/dev/prod to pod validate_url zeby nie bylo z tylka na koncu pliku ze jednak ten gem sie odnosi do wszystkich srodowisk

File: db/migrate/20210806120522_add_nickname_to_user.rb (hunk 2 above)
Comment (grabomarcin) [reply db/migrate/20210806120522_add_nickname_to_user.rb:3]: I'll change that.

**Review by mirecki — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 3 above)
Comment (grabomarcin) [reply Gemfile]: Fixed.

**Review by mirecki — APPROVED** [review]

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+  # == Extensions ===========================================================
+
+  include ActiveModel::Validations
+
+  # == Relationships ========================================================
+
+  # == Validations ==========================================================
+
+  validates :nickname, uniqueness: true, presence: true
```
Comment (h4b00) [inline app/models/user.rb:25]: Add validation for uniqueness email & presence

File: spec/factories/users.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :user do
+
```
Comment (h4b00) [inline spec/factories/users.rb:3]: Why empty?

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  # == Attributes ===========================================================
+
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  # == Extensions ===========================================================
+
+  include ActiveModel::Validations
```
Comment (h4b00) [inline app/models/user.rb:19]: We don't need this I think.

File: app/models/user.rb (hunk 1 above)
Comment (h4b00) [reply app/models/user.rb:14]: let's add this later

**Review by h4b00 — CHANGES_REQUESTED** [review]

File: app/models/user.rb [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+         :recoverable, :rememberable, :validatable
+
+  # == Extensions ===========================================================
+
+  # == Relationships ========================================================
+
+  # == Validations ==========================================================
+
+  validates :nickname, uniqueness: true, presence: true
+  validates :email, uniqueness: true, presence: true
```
Comment (h4b00) [inline app/models/user.rb:24]:
```suggestion
  validates :email, :nickname, uniqueness: true, presence: true
```

File: spec/factories/users.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :user do
+    email { Faker::Internet.email }
+    nickname { Faker::Internet.username }
```
Comment (h4b00) [inline spec/factories/users.rb:4]: Do not use faker to generate unique fields.

Use https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#sequences

File: spec/factories/users.rb [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+    password { Faker::Internet.password }
```
Comment (h4b00) [inline spec/factories/users.rb:5]: Same here ☝️

File: spec/models/user_spec.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (h4b00) [inline spec/models/user_spec.rb:4]: Missing validations specs.

File: spec/rails_helper.rb [hunk 11]
```diff
@@ -33,6 +33,8 @@
 RSpec.configure do |config|
   config.include FactoryBot::Syntax::Methods
 
+  config.include Devise::Test::ControllerHelpers, type: :controller
```
Comment (h4b00) [inline spec/rails_helper.rb:36]: 👍

**Review by h4b00 — CHANGES_REQUESTED** [review]

**Review by Narzew — APPROVED** [review]

**Review by PatrykTyszka — APPROVED** [review]

**Review by h4b00 — APPROVED** [review]
LGTM

**Pull request merged**